src/app/SimpleSubscriptionResumptionStorage.cpp

```diff
@@ -81,173 +81,229 @@ CHIP_ERROR SimpleSubscriptionResumptionStorage::SaveIndex(const SubscriptionInde
     return CHIP_NO_ERROR;
 }
 
+#define DeleteIndexAndReturnLogErrorOnFailure(expr)                                                                                                 \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        auto __err = (expr);                                                                                                       \
+        if (!::chip::ChipError::IsSuccess(__err))                                                                                  \
+        {                                                                                                                          \
+mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName()); \
+ChipLogError(DataManagement, "%s at %s:%d", ErrorStr(__err), __FILE__, __LINE__);                                        \
+            return __err;                                                                                                          \
+        }                                                                                                                          \
+    } while (false)
+
 CHIP_ERROR SimpleSubscriptionResumptionStorage::LoadIndex(SubscriptionIndex & index)
+{
+    return LoadIndex(index, 0);
+}
+
+CHIP_ERROR SimpleSubscriptionResumptionStorage::LoadIndex(SubscriptionIndex & index, size_t allocateExtraSpace)
 {
     Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
     backingBuffer.Calloc(MaxIndexSize());
+    if (backingBuffer.Get() == nullptr)
+    {
+        mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName());
+        return CHIP_ERROR_NO_MEMORY;
+    }
 
     uint16_t len = static_cast<uint16_t>(MaxIndexSize());
 
     index.mSize = 0;
     if (mStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName(), backingBuffer.Get(), len) !=
         CHIP_NO_ERROR)
     {
+        mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName());
         return CHIP_NO_ERROR;
     }
 
     TLV::ScopedBufferTLVReader reader(std::move(backingBuffer), len);
 
-    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    DeleteIndexAndReturnLogErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
```

Comment:
Suggest just doing this on exit rather than every site. We do not need this level of logging. We don't do it elsewhere